things/database.py

```diff
@@ -622,15 +635,14 @@ def make_filter(column, value):
     ''
     """
     default = f"AND {column} = '{escape_string(str(value))}'"
-    # special options
     return {
         None: "",
         False: f"AND {column} IS NULL",
         True: f"AND {column} IS NOT NULL",
     }.get(value, default)
 
 
-def make_date_filter(date_column, offset):
+def make_date_filter(date_column: str, offset) -> str:
```

Comment:
I think this function is getting too complex. I find it hard to read.
Maybe split it out into two functions?

- `make_date_range_filter` (`"3d"`, `"2w"`, `"5y"`)
- `make_date_filter` (`True`, `False`, `"future"`, `"past"`)

Reply:
Personal gut feeling: keep it. These functions would be very similar in what they want to achieve and the current version is around 20 lines of code. But also fine splitting it.

Reply:
I vote for splitting. It seems hard to read, especially for somebody who's never seen this code before.

A date range filter (specified by an offset) and a date filter (True, False, "past", "future) seems more readable for me.

Reply:
One can always merge again if it seems too redundant. I much prefer many smaller functions.

Reply:
Created https://github.com/thingsapi/things.py/issues/82 for this

Reply:
Why break it out into a separate PR?

The proposal was to rename `make_date_filter` to `make_date_range_filter`, but leave it as-is. Then, to add a new function called `make_date_filter` which operates on `True, False, None, "future", "past"`.

Reply:
Mh:
- In case of None: return ""
- In case of True/False: return make_filter
- In case of Future/Past: set offset to 0 and specify operator

The new method would duplicate the actual code
```python
column_datetime = f"datetime({date_column}, 'unixepoch', 'localtime')"
offset_datetime = f"datetime('now', 'localtime', '{modifier}')"  # type: ignore
```
Wouldn't it?

Reply:
For the case of "future" and "past", my thinking was to call `make_date_range_filter` from within `make_date_filter`. Or.. what do you think?

@AlexanderWillner I think we're almost there with this review. I look forward when we can merge this in. :)